internal/session/serve.go

```diff
@@ -87,24 +88,43 @@ func (s *Session) serve() {
 		case "sethealthy":
 			log.Logger.Infow("sethealthy received", "components", payload.Components)
 			for _, componentName := range payload.Components {
-				if componentName != "accelerator-nvidia-error-xid" {
-					continue
-				}
-				rawComponent, err := components.GetComponent("accelerator-nvidia-error-xid")
-				if err != nil {
-					log.Logger.Errorw("failed to get component", "error", err)
-					continue
-				}
-				if watchable, ok := rawComponent.(*metrics.WatchableComponentStruct); ok {
-					if component, ok := watchable.Component.(*xid.XIDComponent); ok {
-						if err = component.SetHealthy(); err != nil {
-							log.Logger.Errorw("failed to set xid healthy", "error", err)
+				switch componentName {
+				case "accelerator-nvidia-error-xid":
+					rawComponent, err := components.GetComponent("accelerator-nvidia-error-xid")
+					if err != nil {
+						log.Logger.Errorw("failed to get component", "error", err)
+						continue
+					}
+					if watchable, ok := rawComponent.(*metrics.WatchableComponentStruct); ok {
+						if component, ok := watchable.Component.(*xid.XIDComponent); ok {
+							if err = component.SetHealthy(); err != nil {
+								log.Logger.Errorw("failed to set xid healthy", "error", err)
+							}
```

Comment:
IMO every component should be watchable, and I thought there was already a `Watchable` interface. However, it seems that such an interface doesn't actually exist. It appears that we are using `WatchableComponentStruct` to wrap other components, and WatchableComponentStruct overrides the States method to update component health metrics. Is it possible to remove this wrapper to make the interfaces clearer and improve readability? This would also simplify the code and make it easier to maintain.
/cc @gyuho @cardyok

Reply:
Yeah, the interface/unwrap being too complicated... If this is already tested, we can do the refactoring in a separate PR. @cardyok